lib/AST/ASTContext.cpp

```diff
@@ -426,6 +426,9 @@ struct ASTContext::Implementation {
   /// Singleton used to cache the import graph.
   swift::namelookup::ImportCache TheImportCache;
 
+  /// Cache of availability macros parsed from the command line.
+  AvailabilityMacroMap TheAvailabilityMacroCache;
```

Comment:
Could we use the request-evaluator here rather than add the cache directly to the ASTContext?

Reply:
Other requests seem to use the ASTContext itself. I'll see to use that.